src/any_agent/serving/a2a/agent_executor.py

```diff
@@ -1,4 +1,20 @@
-from typing import TYPE_CHECKING, override
+import sys
+
+PYTHONEGT312 = sys.version_info >= (3, 12)
+
+from typing import TYPE_CHECKING
+
+if PYTHONEGT312:
+    from typing import override
+else:
+    from collections.abc import Callable
+    from typing import Any, TypeVar
+
+    F = TypeVar("F", bound=Callable[..., Any])
+
+    def override(func: F, /) -> F:  # noqa: D103
+        return func
```

Comment:
I have no idea how this works but I guess if the integration and unit tests work then all good. Is this the implementation in python 3.12 copied pasted here? Would be maybe nice to have a short comment explaining what's going on here.

Reply:
Basically, we define a decorator that doesn't decorate XD Trying to define different classes depending on the python version was much messier. The funny looking stuff is just needed for type checking: we accept any function signature, but we ensure that we return something with that same signature. 

I'll include this as a comment. I am running the integration locally, I'll probably use a separate PR to write the GH tasks for v3.12 and v3.11 integration tests.